plugins/modules/nmcli.py

```diff
@@ -1920,10 +1942,20 @@ def connection_options(self, detect_change=False):
                 raise NmcliModuleError('type is macvlan but all of the following are missing: macvlan')
         elif self.type == 'wireguard':
             if self.wireguard:
-                for name, value in self.wireguard.items():
-                    options.update({
-                        'wireguard.%s' % name: value,
-                    })
+                options.update({
+                    'fwmark':self.wireguard['fwmark'],
+                    'ip4-auto-default-route': self.wireguard['ip4_auto_default_route'],
+                    'ip6-auto-default-route': self.wireguard['ip6_auto_default_route'],
+                    'listen-port': self.wireguard['listen_port'],
+                    'mtu': self.wireguard['mtu'],
+                    'peer-routes': self.wireguard['peer_routes'],
+                    'private-key': self.wireguard['private_key'],
+                    'private-key-flags': self.wireguard['private_key_flags'],
+                    'peer-public-key': self.wireguard['peer_public_key'],
+                    'peer-allowed-ips': self.wireguard['peer_allowed_ips'],
+                    'peer-endpoint': self.wireguard['peer_endpoint'],
+                    'peer-persistent-keepalive': self.wireguard['peer_persistent_keepalive'],
```

Comment:
This looks worng to me. 1) it only hadles *some* of the entries in `self.wireguard`, and not all the others the user might have supplied. 2) The `wireguard.` prefix is not there.